graph/builder.go

```diff
@@ -0,0 +1,1837 @@
+		case msg.ChannelFlags&lnwire.ChanUpdateDirection == 1:
+
+			// Ignore outdated message.
+			if !edge2Timestamp.Before(msg.LastUpdate) {
+				return newErrf(ErrOutdated, "Ignoring "+
+					"outdated update (flags=%v|%v) for "+
+					"known chan_id=%v", msg.MessageFlags,
+					msg.ChannelFlags, msg.ChannelID)
+			}
+		}
+
+		// Now that we know this isn't a stale update, we'll apply the
+		// new edge policy to the proper directional edge within the
+		// channel graph.
+		if err = b.cfg.Graph.UpdateEdgePolicy(msg, op...); err != nil {
+			err := errors.Errorf("unable to add channel: %v", err)
+			log.Error(err)
+			return err
+		}
+
+		log.Tracef("New channel update applied: %v",
+			newLogClosure(func() string { return spew.Sdump(msg) }))
+		b.stats.incNumChannelUpdates()
+
+	default:
+		return errors.Errorf("wrong routing update message type")
+	}
+
+	return nil
+}
+
+// fetchFundingTxWrapper is a wrapper around fetchFundingTx, except that it
+// will exit if the router has stopped.
+func (b *Builder) fetchFundingTxWrapper(chanID *lnwire.ShortChannelID) (
+	*wire.MsgTx, error) {
+
+	txChan := make(chan *wire.MsgTx, 1)
+	errChan := make(chan error, 1)
+
+	go func() {
+		tx, err := b.fetchFundingTx(chanID)
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		txChan <- tx
+	}()
+
+	select {
+	case tx := <-txChan:
+		return tx, nil
+
+	case err := <-errChan:
+		return nil, err
+
+	case <-b.quit:
+		return nil, ErrGraphBuilderShuttingDown
+	}
+}
+
+// fetchFundingTx returns the funding transaction identified by the passed
+// short channel ID.
+//
+// TODO(roasbeef): replace with call to GetBlockTransaction? (would allow to
+// later use getblocktxn).
+func (b *Builder) fetchFundingTx(
+	chanID *lnwire.ShortChannelID) (*wire.MsgTx, error) {
+
+	// First fetch the block hash by the block number encoded, then use
+	// that hash to fetch the block itself.
+	blockNum := int64(chanID.BlockHeight)
+	blockHash, err := b.cfg.Chain.GetBlockHash(blockNum)
+	if err != nil {
+		return nil, err
+	}
+	fundingBlock, err := b.cfg.Chain.GetBlock(blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	// As a sanity check, ensure that the advertised transaction index is
+	// within the bounds of the total number of transactions within a
+	// block.
+	numTxns := uint32(len(fundingBlock.Transactions))
+	if chanID.TxIndex > numTxns-1 {
+		return nil, fmt.Errorf("tx_index=#%v "+
+			"is out of range (max_index=%v), network_chan_id=%v",
+			chanID.TxIndex, numTxns-1, chanID)
+	}
+
+	return fundingBlock.Transactions[chanID.TxIndex].Copy(), nil
+}
+
+// routingMsg couples a routing related routing topology update to the
+// error channel.
+type routingMsg struct {
+	msg interface{}
+	op  []batch.SchedulerOption
+	err chan error
+}
+
+// ApplyChannelUpdate validates a channel update and if valid, applies it to the
+// database. It returns a bool indicating whether the updates were successful.
+func (b *Builder) ApplyChannelUpdate(msg *lnwire.ChannelUpdate) bool {
```

Comment:
oh i see what you mean now. I think these do 2 different things at the moment though: 

- `(s *server) applyChannelUpdate` sends the update to the gossiper and takes a "useAlias" param
- `(b *Builder) ApplyChannelUpdate` does some validation and persists. 

Perhaps they can be merged eventually but let's keep as is for now just to avoid the behaviour change here?